pkg/apis/pingcap/v1alpha1/tidb_config_wraper.go

```diff
@@ -78,7 +78,11 @@ func unmarshalJSON(data []byte, x interface{}) (g *config.GenericConfig, err err
 	case string:
 		tomlData = []byte(s)
 	case map[string]interface{}:
-		err = json.Unmarshal(data, &x)
+		if x == nil {
```

Comment:
`json.Unmarshal` will panic if the `x == nil`. Do we need to check it here?

Reply:
https://github.com/pingcap/tidb-operator/pull/3351/commits/1a977112e0b07977cc83190e04c27ff0dd85d2a2 PTAL